dashboard/api.py

```diff
@@ -0,0 +1,227 @@
+
+    if run_status.status in (CourseRunStatus.NOT_ENROLLED, CourseRunStatus.NOT_PASSED):
+        return format_course_for_dashboard(
+            course.get_next_run(),
+            CourseStatus.OFFERED if course.get_next_run() is not None else CourseStatus.NOT_OFFERED,
+            course
+        )
+    elif run_status.status == CourseRunStatus.GRADE:
+        return format_course_for_dashboard(run_status.course_run, CourseStatus.CURRENT_GRADE, course)
+    # check if we need to check the certificate
+    elif run_status.status == CourseRunStatus.READ_CERT:
+        # TODO: here should go the logic to check the actual certificate  # pylint: disable=fixme
+        #       for now just assume not passed and return the next available course run
+        # this is the case when the certificate says the user did not pass the course
+        return format_course_for_dashboard(
+            course.get_next_run(),
+            CourseStatus.OFFERED if course.get_next_run() is not None else CourseStatus.NOT_OFFERED,
+            course
+        )
+    elif run_status.status == CourseRunStatus.WILL_ATTEND:
+        return format_course_for_dashboard(run_status.course_run, CourseStatus.CURRENT_GRADE, course)
+    elif run_status.status == CourseRunStatus.UPGRADE:
+        return format_course_for_dashboard(run_status.course_run, CourseStatus.UPGRADE, course)
+    # this should never happen, but put a default behavior just in case
+    log.critical(
+        'course %s for user %s has status %s',
+        run_status.course_run.edx_course_key,
+        user.username,
+        run_status.status
+    )
+    return format_course_for_dashboard(
+        course.get_next_run(),
+        CourseStatus.OFFERED if course.get_next_run() is not None else CourseStatus.NOT_OFFERED,
+        course
+    )
+
+
+def get_status_for_courserun(course_run, user_enrollments):
+    """
+    Checks the status of a course run for a user given her enrollments
+
+    Args:
+        course_run (CourseRun): a course run
+        user_enrollments (Enrollments): the user enrollments
+
+    Returns:
+        CourseRunStatus: an object representing the run status for the user
+    """
+    if not user_enrollments.is_enrolled_in(course_run.edx_course_key):
+        return CourseRunUserStatus(CourseRunStatus.NOT_ENROLLED, course_run)
+    course_enrollment = user_enrollments.get_enrollment_for_course(course_run.edx_course_key)
+    if course_enrollment.is_verified:
+        if course_run.is_current:
+            status = CourseRunStatus.GRADE
+        elif course_run.is_past:
+            status = CourseRunStatus.READ_CERT
+        elif course_run.is_future:
+            status = CourseRunStatus.WILL_ATTEND
+    else:
+        if course_run.is_current or course_run.is_future:
+            status = CourseRunStatus.UPGRADE
+        elif course_run.is_past:
+            status = CourseRunStatus.NOT_PASSED
+    return CourseRunUserStatus(
+        status=status,
```

Comment:
Can you define `status = None` above? My IDE is complaining that `status` might not be defined in all cases